server/src/controllers/cartAndOrder.ts

```diff
@@ -0,0 +1,27 @@
+import { NextFunction, Request, Response } from "express";
+import { addProductToCartService, getCartForUserService, placeOrderService, updateCartService } from "../services/cartAndOrders";
+import { cartModel, orderModel } from "../models/types";
+
+export const getCartForUser = async (req: Request, res: Response) => {
+    const user_email = req.query.userEmail as string;
+    let cart = await getCartForUserService(user_email);
+    return res.send(cart);
+}
+
+export const addProductToCart = async (req: Request, res: Response) => {
+    const product: cartModel = new cartModel(req.body);
+    await addProductToCartService(product);
+    return res.send("product added to cart");
```

Comment:
We generally send the added product back in response with generated ID, in POST req's.